src/hooks/useCountdown.js

```diff
@@ -4,24 +4,26 @@ export default function useCountdown(deadline) {
     const [timeLeft, setTimeLeft] = useState(getTimeLeft(deadline));
 
     useEffect(() => {
+        if (timeLeft.totalInSeconds === 0) return;
+
         const interval = setInterval(() => {
             setTimeLeft(getTimeLeft(deadline));
         }, 1000);
 
         return () => clearInterval(interval);
-    }, [deadline]);
+    }, [deadline, timeLeft.totalInSeconds]);
```

Comment:
Is it really required to put `timeleft.totalInSeconds` to make it stop when it reaches to zero? I thought `if (...) return;` would do the job.

Reply:
@0xferit,  including `timeLeft.totalInSeconds` in the dependency array is necessary to reevaluate the effect whenever totalInSeconds reaches zero. 
`if (...) return;` is used to stop the interval from running when the countdown reaches zero. However, without including _totalInSeconds_ in the dependency array, the effect will not be re-run when the time is up. ( meaning that the interval will not be cleared, and the component will continue to update)

Reply:
Makes sense; thanks for explaining.